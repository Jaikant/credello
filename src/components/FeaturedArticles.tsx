import * as React from 'react';
import styled from 'styled-components';
import { TextBold, TextMedium } from 'components/Typography';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';

const InlineDiv = styled.div`
  display: flex;
`;

const Category = styled(TextBold)`
  margin-right: 8px;
  color: rgb(92, 219, 149);
`;

const UpdatedOn = styled(TextBold)`
  color: #808080;
`;

const TextBody = styled(TextMedium)`
  line-height: 1.43;
  letter-spacing: 0.01071em;
`;

const StyledCard = styled(Card)`
  max-width: 345px;
`;

const StyledCardMedia = styled(CardMedia)`
  height: 140px;
`;

function MediaCard(props) {
  return (
    <StyledCard>
      <CardActionArea>
        <StyledCardMedia
          image={require(`../assets/images/${props.image}`)}
          title={props.title}
        />
        <CardContent>
          <InlineDiv>
            <Category size="12px" mb="8px">
              {props.category}
            </Category>
            <UpdatedOn size="12px" mb="8px">
              {props.updated}
            </UpdatedOn>
          </InlineDiv>
          <TextBold size="18px" mb="8px">
            {props.title}
          </TextBold>
          <TextBody size="14px" color="#808080">
            {props.content}
          </TextBody>
        </CardContent>
      </CardActionArea>
    </StyledCard>
  );
}

//https://github.com/cyrilwanner/next-optimized-images/issues/141#issuecomment-623556804
MediaCard.defaultProps = {
  image: 'featured.png',
  category: 'DEBT 101',
  updated: ' | Mar 19, 2020',
  title: `Whats a Normal Amount of Debt... and What's Too Much?`,
  content: `If you're not comfortable about your debt, you're not alone. 45% of adults admint debt makes them
  feel anxious. Read on to see what debt looks like for an...`,
};

const Block = () => {
  return (
    <>
      <TextBold size="24px" mb="8 px">
        Featured articles
      </TextBold>
      <MediaCard />
    </>
  );
};

Block.defaultProps = {};

export default Block;
