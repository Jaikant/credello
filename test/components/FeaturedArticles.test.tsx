import React from 'react';
import { render } from '../testUtils';
import FeaturedArticles from '../../src/components/FeaturedArticles';
import 'jest-styled-components';

describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<FeaturedArticles />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
