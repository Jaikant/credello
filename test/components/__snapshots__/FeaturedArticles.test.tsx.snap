// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  font-family: Roboto-Bold;
  font-weight: 700;
  font-size: 24px;
  line-height: normal;
  margin: 0;
  margin-bottom: 16px;
  margin-top: 0;
  color: #02254d;
  text-align: initial;
}

.c9 {
  font-family: Roboto-Bold;
  font-weight: 700;
  font-size: 18px;
  line-height: 24px;
  margin: 0;
  margin-bottom: 8px;
  margin-top: 0;
  color: inherit;
  text-align: initial;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  font-family: Roboto-Bold;
  font-weight: 700;
  font-size: 12px;
  line-height: 16px;
  margin: 0;
  margin-bottom: 8px;
  margin-top: 0;
  color: inherit;
  text-align: initial;
  margin-right: 8px;
  color: rgb(92,219,149);
}

.c7 {
  font-family: Roboto-Regular;
  font-weight: 400;
  font-size: 12px;
  line-height: 16px;
  webkit-font-smoothing: antialiased;
  margin: 0;
  margin-bottom: 8px;
  margin-top: 0;
  color: inherit;
  text-align: initial;
  color: #808080;
}

.c10 {
  font-family: Roboto-Regular;
  font-weight: 400;
  font-size: 16px;
  line-height: 20px;
  webkit-font-smoothing: antialiased;
  margin: 0;
  margin-bottom: 0;
  margin-top: 0;
  color: #6c7f87;
  text-align: initial;
  -webkit-letter-spacing: 0.01071em;
  -moz-letter-spacing: 0.01071em;
  -ms-letter-spacing: 0.01071em;
  letter-spacing: 0.01071em;
}

.c2 {
  max-width: 500px;
}

.c2.MuiPaper-root {
  box-shadow: none;
}

.c3 {
  height: 160px;
  border-radius: 4px;
}

.c4.MuiCardContent-root {
  padding-right: 0;
  padding-left: 0;
}

.c8 {
  max-width: 278px;
}

.c0 {
  display: grid;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-top: 34px;
  padding-bottom: 40px;
}

@media (min-width:768px) {
  .c1 {
    font-size: 24px;
  }
}

@media (min-width:1224px) {
  .c1 {
    line-height: 1.5;
    font-size: 24px;
  }
}

@media (min-width:768px) {
  .c9 {
    font-size: 18px;
  }
}

@media (min-width:1224px) {
  .c9 {
    line-height: 1.5;
    font-size: 18px;
  }
}

@media (min-width:768px) {
  .c6 {
    font-size: 12px;
  }
}

@media (min-width:1224px) {
  .c6 {
    line-height: 1.5;
    font-size: 12px;
  }
}

@media (min-width:768px) {
  .c7 {
    font-size: 12px;
  }
}

@media (min-width:1224px) {
  .c7 {
    font-size: 12px;
  }
}

@media (min-width:768px) {
  .c10 {
    font-size: 16px;
  }
}

@media (min-width:1224px) {
  .c10 {
    font-size: 16px;
  }
}

@media (min-width:768px) {
  .c8 {
    max-width: auto;
  }
}

@media (min-width:1224px) {

}

<div
    class="c0"
  >
    <h3
      class="c1"
      color="#02254d"
    >
      Featured articles
    </h3>
    <div
      class="MuiPaper-root MuiCard-root c2 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root c3"
          style="background-image: url(test-file-stub);"
          title="Whats a Normal Amount of Debt... and What's Too Much?"
        />
        <div
          class="MuiCardContent-root c4"
        >
          <div
            class="c5"
          >
            <h3
              class="c6"
            >
              DEBT 101
            </h3>
            <div
              class="Typography__Text-sc-12f2dd0-0 c7"
            >
               | Mar 19, 2020
            </div>
          </div>
          <div
            class="c8"
          >
            <h3
              class="c9"
            >
              Whats a Normal Amount of Debt... and What's Too Much?
            </h3>
          </div>
          <div
            class="Typography__Text-sc-12f2dd0-0 c10"
            color="#6c7f87"
          >
            If you're not comfortable about your debt, you're not alone. 45% of adults admint debt makes them
  feel anxious on a monthly basis
          </div>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
